### PR TITLE
feat(storages-s3): handle 403 errors

### PR DESCRIPTION
**Pull request by radimsuckr**
Throw a PermissionError when S3 returns a 403 error.

### CODE REVIEW

File: chamber/storages/boto3.py [hunk 1]
```diff
@@ -46,6 +47,13 @@ def _clean_name(self, name):
         # pathlib support
         return super()._clean_name(str(name))
 
+    def _open(self, name, mode='rb'):
+        try:
+            return super()._open(name, mode='rb')
+        except ClientError as ex:
+            if ex.response['Error']['Code'] == '403':
+                raise PermissionError(f'Cannot open file "{name}": {ex.response["Error"]["Message"]}')
```
Comment (matllubos) [inline chamber/storages/boto3.py:55]: Where is imported PermissionError I cannot find it.

Comment (radimsuckr) [reply chamber/storages/boto3.py:55]: It's a built-in exception. https://docs.python.org/3/library/exceptions.html#PermissionError

Comment (matllubos) [reply chamber/storages/boto3.py:55]: You are right I miss this exception with python 3 upgrade from python 2. It make sense

You should reraise the client error if exception is not 403.

Comment (radimsuckr) [reply chamber/storages/boto3.py:55]: I added exception reraising

**Pull request merged**